### ISSUE TITLE
marktree_itr_next_skip: Assertion `itr->x->ptr[itr->i]->parent == itr->x' failed.

### ISSUE THREAD

**Issue by Kamilcuk** [neovim/neovim#27153]
### Problem

Assertion triggers on neovim master on newest coc-nvim completion when using atlassian Python API. 

```
nvim: /home/kcukrowski/.cache/,nvim/neovim/src/nvim/marktree.c:1529: marktree_itr_next_skip: Assertion `itr->x->ptr[itr->i]->parent == itr->x' failed.
```

```
$ nvim -V1 -v
NVIM v0.10.0-dev-2164+gdc466f9a6
Build type: RelWithDebInfo
LuaJIT 2.1.1705947465
Compilation: /usr/bin/cc -O2 -g -Og -g -flto -fno-fat-lto-objects -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wsh
adow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fsigned-char -fstack-protector-stron
g -fno-common -Wno-unused-result -Wimplicit-fallthrough -fdiagnostics-color=always  -DUNIT_TESTING -DHAVE_UNIBILIUM -D_GNU_SOURCE -DINCLUDE_GENERAT
ED_DECLARATIONS -DNVIM_TS_HAS_SET_MAX_START_DEPTH -I/home/kcukrowski/.cache/,nvim/neovim/.deps/usr/include/luajit-2.1 -I/home/kcukrowski/.cache/,nv
im/neovim/.deps/usr/include -I/home/kcukrowski/.cache/,nvim/neovim/build/src/nvim/auto -I/home/kcukrowski/.cache/,nvim/neovim/build/include -I/home
/kcukrowski/.cache/,nvim/neovim/build/cmake.config -I/home/kcukrowski/.cache/,nvim/neovim/src -I/usr/include

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/home/kcukrowski/.cache/,nvim/install/share/nvim"

Run :checkhealth for more info
```

This does _not_ happen with the exact same environment with /snap/bin/nvim verison 0.9.4 installed from snap.

### Steps to reproduce

1. pip install atlassian-python-api pynvim
2. Run nvim with coc-nvim
3. Write the following python script:
```
import atlassian
conf = Confluence()
conf.
```
4. Position cursor position after `conf.`
5. Trigger coc completion with ctrl+space in my config.
6. Neovim will shut down with the error above.

### Expected behavior

No assertions!

### Neovim version (nvim -v)

NVIM v0.10.0-dev-2164+gdc466f9a6 Build type: RelWithDebInfo LuaJIT 2.1.1705947465 Run "nvim -V1 -v" for more info

### Vim (not Nvim) behaves the same?

no, coc-nvim is for nvim

### Operating system/version

WSL2 Ubuntu 22.04.3 LTS

### Terminal name/version

Alatricitty and tmux3

### $TERM environment variable

xterm-256color

### Installation

Compiled from source

Thanks for making neovim, it is great!

**Comment by zeertzjq** [comment]
Can you try the latest commit?

**Comment by Kamilcuk** [comment]
Hi! Thanks, works with v0.10.0-dev-1630+gc126a3756 . I'll be checking master next time. ;)

**Comment by wookayin** [comment]
Yes this is already fixed by (Duplicate of) #27137.
